python/ray/util/ray_executor.py

```diff
@@ -0,0 +1,237 @@
+import itertools
+import time
+from concurrent.futures import Executor
+from functools import partial
+
+import ray
+from ray.util.annotations import PublicAPI
+
+
+@PublicAPI(stability="alpha")
+class RayExecutor(Executor):
+    """`RayExecutor` is a drop-in replacement for `ProcessPoolExecutor` and
+    `ThreadPoolExecutor` from `concurrent.futures` but distributes and executes
+    the specified tasks over a Ray cluster instead of multiple processes or
+    threads.
+    """
+
+    def __init__(self, max_workers=None, **kwargs):
+        """Initialises a new RayExecutor instance which distributes tasks over
+        a Ray cluster.
+
+        Args:
```

Comment:
Looks like the init docstring doesn't come up to the documentation. Can you move it to the class level docstring?